src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -114,15 +114,6 @@ class SearchBarUI extends Component<SearchBarProps, State> {
         query: nextProps.query.query,
         language: nextProps.query.language,
       };
-    } else if (
```

Comment:
I'm assuming this logic moved to https://github.com/elastic/kibana/pull/61611/files#diff-6fce6ef7c306a45fc9b878e0bcae061bR356 - what was the reason for this change?

Reply:
yes

Reply:
Can you clarify why you moved it over?

Reply:
Ah, sorry, missed the question. This whole logic of setting state from props is very convoluted. We have a query and a saved query, which from the point of this component is basically a query (it goes in the query bar). Then there is props, nextProps, prevState.currentProps... 🤯 In my original PR I wanted to get rid of this logic altogether and move some of this in `componentDidMount`, but that was another can of worms.
If I remember correctly, I got rid of this `else` bit, cause things seem to work fine without it and it's less convoluted with out. I don't think it's related to clearing up `savedQuery`, that was another issue.

Reply:
Thanks for the explanation - I thought this was fixing some kind of buggy behavior and couldn't figure it out. I agree that it's easier to follow this way.

Reply:
@majagrubic @flash1293 

A saved query is entirely different from a `query`.
I know that's very bad naming, but a saved query is a name for filters + query string + timefilter, while a `query` is just a `query string` + `query language` configuration.

So moving this to the saved query logic is definitely not doing the same thing. Would you mind exploring this in a separate PR?

Reply:
I understand query string is different from saved query, but from the point of view of `QueryStringInput` component it *is* the same - ie that component is only interested in `query string` + `language`.